nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -89,9 +89,15 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         # Freeze all GPT model weights for prompt-tuning/p-tuning
         self.frozen_model.freeze()
         self.tokenizer = self.frozen_model.tokenizer
-        self.float_type = self.frozen_model.model.language_model.encoder.layers[0].dtype
+
+        if self.frozen_model.megatron_amp_o2:
```

Comment:
Can we get this from the `precision` attribute of `frozen_model` instead of trying to access the weights? T5 models for example will use `enc_dec_model` instead of just `model` and may have extra nesting and we'll have to figure this out for every kind of model for prompt learning of other models.